src/highdicom/sr/templates/tid3700.py

```diff
@@ -0,0 +1,1213 @@
+    """:dcm:`TID 3702 <part16/sect_TID_3702.html>`
+    Prior ECG Study
+    """
+
+    def __init__(
+        self,
+        comparison_with_prior_study_done: Union[CodedConcept, Code],
+        procedure_datetime: Optional[Union[str, datetime, DT]] = None,
+        procedure_study_instance_uid: Optional[UIDRefContentItem] = None,
+        prior_report_for_current_patient: Optional[CompositeContentItem] = None,
+        source_of_measurement: Optional[WaveformContentItem] = None
+    ):
+        item = ContainerContentItem(
+            name=codes.LN.PriorProcedureDescriptions,
+            template_id='3702',
+            relationship_type=RelationshipTypeValues.CONTAINS
+        )
+        content = ContentSequence()
+        comparison_with_prior_study_done_item = CodeContentItem(
+            name=CodedConcept(
+                value='122140',
+                meaning='Comparison with Prior Study Done',
+                scheme_designator='DCM'
+            ),
+            value=comparison_with_prior_study_done,
+            relationship_type=RelationshipTypeValues.CONTAINS
+        )
+        content.append(comparison_with_prior_study_done_item)
+        if procedure_datetime is not None:
+            if not isinstance(procedure_datetime, (str, datetime, DT)):
+                raise TypeError(
+                    'Argument "procedure_datetime" must have type ' +
+                    'str, datetime.datetime or DT.'
+                )
+            procedure_datetime_item = DateTimeContentItem(
+                name=codes.DCM.ProcedureDatetime,
+                value=procedure_datetime,
+                relationship_type=RelationshipTypeValues.CONTAINS
+            )
+            content.append(procedure_datetime_item)
+        if procedure_study_instance_uid is not None:
+            if not isinstance(procedure_study_instance_uid,
+                              UIDRefContentItem):
+                raise TypeError(
+                    'Argument "procedure_stdy_instance_uid" must have ' +
+                    'type UIDRefContentItem.'
+                )
+            content.append(procedure_study_instance_uid)
+        if prior_report_for_current_patient is not None:
+            if not isinstance(prior_report_for_current_patient,
+                              CompositeContentItem):
+                raise TypeError(
+                    'Argument "prior_report_for_current_patient" must ' +
+                    'have type CompositeContentItem.'
+                )
+            content.append(prior_report_for_current_patient)
+        if source_of_measurement is not None:
+            if not isinstance(source_of_measurement, WaveformContentItem):
+                raise TypeError(
+                    'Argument "source_of_measurement" must have type ' +
+                    'WaveformContentItem.'
+                )
+            content.append(source_of_measurement)
+        if len(content) > 0:
+            item.ContentSequence = content
```

Comment:
Check is unnecessary since some items are required. This occurs in several places. I haven't left a comment on all of them